### PR TITLE
[Feature] Add logprob support for overlap spec v2 (EAGLE)

### PR DESCRIPTION
**Pull request by Dayuxiaoshui**
[Feature] Overlap Spec Support #11762

## Motivation

This PR adds **logprob support for overlap spec v2 (EAGLE)**. When `return_logprob=True`, the server should return `output_token_logprobs` (and optionally `output_top_logprobs`) for each generated token. Previously this path did not populate logprobs on the verify branch; this change fills them from the verify-step logits so that clients get correct per-token logprobs when using speculative decoding with spec v2.

## Modifications

- **logprob.py**  
  Add `add_output_logprobs_for_spec_v2(batch, logits_output, predict, accept_length, accept_index, stride)`, which takes the full verify logits, selects positions according to `accept_index`, applies temperature and log_softmax, and writes results into `req.output_token_logprobs_val/idx` and (when requested) top_logprobs. Handles `batch.top_logprobs_nums is None`.

- **multi_layer_eagle_worker_v2.py**  
  In the verify path, after `verify_input.sample()` and when `batch.return_logprob` and not idle, call `add_output_logprobs_for_spec_v2(...)` so that accepted tokens get logprobs before building `next_draft_input`.

- **eagle_worker_v2.py**  
  Same as above for the single-layer EAGLE v2 verify path.

- **test_logprob_spec_v2.py** (new)  
  Unit tests for `add_output_logprobs_for_spec_v2`: empty accept_index early return, single/multiple tokens per request, ordering, and `top_logprobs_nums is None` (uses zeros).

- **test_eagle_infer_beta.py**  
  Integration tests: `test_logprob_spec_v2` (return_logprob, check `output_token_logprobs` length and shape) and `test_logprob_spec_v2_top_logprobs` (with `top_logprobs_num`).

## Accuracy Tests

N/A — no changes to kernel or model forward; only logprob wiring and tests.

## Checklist

- [ ] Format your code according to the [Format code with pre-commit](https://docs.sglang.io/developer_guide/contribution_guide.html#format-code-with-pre-commit).
- [ ] Add unit tests according to the [Run and add unit tests](https://docs.sglang.io/developer_guide/contribution_guide.html#run-and-add-unit-tests).
- [ ] Update documentation according to [Write documentations](https://docs.sglang.io/developer_guide/contribution_guide.html#write-documentations).
- [ ] Provide accuracy and speed benchmark results according to [Test the accuracy](https://docs.sglang.io/developer_guide/contribution_guide.html#test-the-accuracy) and [Benchmark the speed](https://docs.sglang.io/developer_guide/contribution_guide.html#benchmark-the-speed).
- [ ] Follow the SGLang code style [guidance](https://docs.sglang.io/developer_guide/contribution_guide.html#code-style-guidance).

## Review Process

1. Ping Merge Oncalls to start the PR flow. See the [PR Merge Process](https://github.com/sgl-project/sglang/blob/main/.github/MAINTAINER.md#pull-request-merge-process).
2. Get approvals from [CODEOWNERS](https://github.com/sgl-project/sglang/blob/main/.github/CODEOWNERS) and other reviewers.
3. Trigger CI tests with [comments](https://docs.sglang.io/developer_guide/contribution_guide.html#how-to-trigger-ci-tests) or contact authorized users to do so.
   - `/tag-run-ci-label`, `/rerun-failed-ci`, `/tag-and-rerun-ci`
4. After green CI and required approvals, ask Merge Oncalls to merge.

### CODE REVIEW

**Comment by Dayuxiaoshui** [comment]
@hnyls2002

File: python/sglang/srt/layers/utils/logprob.py [hunk 1]
```diff
@@ -405,3 +405,81 @@ def add_output_logprobs_for_spec_v1(
...
+        logprobs = torch.nn.functional.log_softmax(
+            accepted_logits / temperatures.unsqueeze(-1), dim=-1
+        )
+    verified_ids = predict[flat_accepted].long()
+    device = logprobs.device
+    next_token_logprobs_t = logprobs[
+        torch.arange(num_accepted, device=device),
+        verified_ids,
+    ]
+    num_tokens_per_req = accept_length.tolist()
```
Comment (hnyls2002) [inline python/sglang/srt/layers/utils/logprob.py:455]: Why this tolist()? This actually implements sync-manner logprob calculation on spec-v2, which is wrong.

**Review by hnyls2002 — CHANGES_REQUESTED** [review]
